Automattic/wp-calypso
# Support Article: Migrate to useRouteModal to control modal behavior

#### Changes proposed in this Pull Request

* Migrate to `useRouteModal` hook
* Introduce `withRouteModal` to inject `routeModalData` to wrapped component

#### Testing instructions



1. Find a link that opens a support article.
2. Notice that if you click the back button you end up on the page you expect.
3. Open up the link again.
4. Refresh the page. Notice that the modal opens/loads as expected on page load.
5. Close the support article and notice that the query parameter from the url is now gone.



Related to https://github.com/Automattic/wp-calypso/issues/58511, https://github.com/Automattic/wp-calypso/pull/60694
